# Description: Allow user to undo sharing/unsharing a folder without needing to
# close the window beforehand
# Origin: vendor
# Bug-Ubuntu: https://bugs.launchpad.net/bugs/392165
# Forwarded: no
# Author: Chow Loong Jin <hyperair@ubuntu.com>
---

 src/nautilus-share.c |   10 ++++++++++
 1 files changed, 10 insertions(+), 0 deletions(-)


Index: nautilus-share/src/nautilus-share.c
===================================================================
--- nautilus-share.orig/src/nautilus-share.c	2009-09-08 03:30:29.000000000 +0800
+++ nautilus-share/src/nautilus-share.c	2009-09-08 03:30:58.000000000 +0800
@@ -420,6 +420,12 @@
   if (!is_shared)
     restore_saved_permissions (page->path);
 
+  /* update initially shared state, so that we may undo later on */
+  if (retval)
+  {
+    page->was_initially_shared = is_shared;
+    page->is_dirty = FALSE;
+  }
   return retval;
 }
 
@@ -550,6 +556,8 @@
     apply_is_sensitive = page->was_initially_shared;
 
   gtk_widget_set_sensitive (page->button_apply, apply_is_sensitive);
+  gtk_button_set_label (GTK_BUTTON(page->button_apply),
+          _((page->was_initially_shared) ? "Modify _Share" : "Create _Share"));
 }
 
 static void
@@ -917,6 +925,8 @@
     {
       if (page->standalone_window)
 	gtk_widget_destroy (page->standalone_window);
+      else
+        property_page_check_sensitivity (page);
     }
 }
 
